### PR TITLE
[Feature] Support reload class when use VirtualMachine#loadAgent

### PR DESCRIPTION
**Pull request by leozhiyu**
### My Test Evidence

Test Steps

1. Create a test class to dynamic attach target process. eg, RunAttachProfiler.java
2. Run HelloWorldApplication class.
3. Update process id in RunAttachProfiler.java and run RunAttachProfiler class.
4. We can see the result in the picture below.

Before my modification, i can not see the result.

image

### CODE REVIEW

**Comment by leozhiyu** [comment]
Hi @vectorijk , Would you help reivew this?

**Comment by leozhiyu** [comment]
Hi @ CodingCat, would you help review this?

**Comment by leozhiyu** [comment]
Hi @hiboyang , Would you help reivew this?

**Comment by hiboyang** [comment]
> Hi @hiboyang , Would you help review this?

This is a great feature! Thanks for adding it! I do not work in Uber anymore. Maybe @vectorijk  could help to review and merge?

File: src/main/java/com/uber/profiling/AgentImpl.java [hunk 1]
```diff
@@ -76,7 +80,28 @@ public void run(Arguments arguments, Instrumentation instrumentation, Collection
...
+                    .map(ClassMethodArgument::getClassName).collect(Collectors.toSet()));
+
+            tobeReloadClasses.retainAll(loadedClasses);
+
+            tobeReloadClasses.forEach(clazz -> {
+                try {
+                    instrumentation.retransformClasses(Class.forName(clazz));
+                    logger.info("Reload class [" + clazz + "] success.");
+                } catch (Exception e) {
+                    logger.warn("Reload class [" + clazz + "] failed.");
```
Comment (vectorijk) [inline src/main/java/com/uber/profiling/AgentImpl.java:102]: shall we pass and print out the exception in waring logging also?
like `logger.warn("Reload class [" + clazz + "] failed.", e);`

Comment (leozhiyu) [reply src/main/java/com/uber/profiling/AgentImpl.java]: fixed it, thanks.

**Comment by vectorijk** [comment]
lgtm. @leozhiyu thanks for the contribution!

**Pull request merged**